src/libmongoc/tests/TestSuite.c

```diff
@@ -112,6 +113,42 @@ static BSON_ONCE_FUN (_test_suite_ensure_mutex_once)
 }
 
 
+static void
+_handle_signal (int signum)
+{
+   const char *s = "\nProcess was interrupted by the delivery of a signal.\n";
+   const char *sigstr;
+   switch (signum) {
+   case SIGABRT:
+      sigstr = "SIGABRT - Abnormal termination";
+      break;
+   case SIGINT:
+      sigstr = "SIGINT - Interrupted";
+      break;
+   case SIGTERM:
+      sigstr = "SIGTERM - Termination requested";
+      break;
+   case SIGSEGV:
+      sigstr = "SIGSEGV - Access violation";
+      break;
+   default:
+      sigstr = "(Unknown signal delivered)";
+   }
+#ifdef BSON_OS_UNIX
+   /* On POSIX these APIs are signal-safe */
+   write (STDERR_FILENO, s, strlen (s));
+   write (STDERR_FILENO, "  ", 2);
+   write (STDERR_FILENO, sigstr, strlen (sigstr));
+   fsync (STDERR_FILENO);
+#else
+   /* On Windows these APIs are signal-safe */
+   fprintf (stderr, "\n%s\n  %s\n", s, sigstr);
+   fflush (stderr);
+#endif
+   _Exit (signum);
```

Comment:
```suggestion
   exit (signum);
```
What is the motivation for explicitly calling file flush functions and `_Exit ()` rather than delegating to `exit ()`?

Reply:
`exit()` is not signal-safe, but `_Exit()` is signal-safe. `quick_exit()` is also signal-safe (and preferable), but isn't available until C11. (`exit()` will execute arbitrary code that might not be safe in the current process condition e.g. there's an `assert()` that failed and the program invariants are broken, or `SIGINT` might be delivered in the middle of a code block that is holding a global lock). It would be more sophisticated and complete to restore the prior signal handlers and re-`raise()`, but I was experiencing a lot of trouble getting that to work and it isn't yet needed in our test suite.

Reply:
It seems `exit()` is signal-safe, but not _async_-signal-safe (in large part due to the invocation of an arbitrary set of `atexit` functions as you described). Regardless, the reasons you have given are justified.